src/exercises/day-1/for-loops-bonus.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+fn transpose(matrix: &[&[i32]; 3]) -> [[i32; 3]; 3] {
```

Comment:
The `&[&[i32]; 3]` type here should be `&[&[i32]]` to match what the bonus question talks about: a slice-of-slices instead of an array-of-slices.

You will still need to use an owned type for the return value. Instead of a fixed-size array, I would suggest using a `Vec<Vec<i32>>` as the return type. 

However, it will become more and more annoying for you to work with these types: now `pretty_print` no longer works on the result of `transpose` (it will probably take `&[&[i32]]`, but it's not possible to directly convert `Vec<Vec<i32>>` to `&[&[i32]]`).

I think we should instead drop the solution code and add a note to the Markdown file.

Reply:
Hi, agree to change it to `Vec<Vec<i32>>` instead. The leaking method works, but it's not a proper way to do it in real code unless giving it a wrapper that reclaims the memory after `drop`. But this is superfluous as `Vec` already serves this purpose

Reply:
Yeah, agreed — we want to show people nice code, so no leaking please :smile: